### ISSUE TITLE
Allow specifying some allowed "extraneous-dependencies"

### ISSUE THREAD

**Issue by coopy** [benmosher/eslint-plugin-import#455]
When using the `import/no-extraneous-dependencies` rule, while having a webpack `resolve.alias` configured so as to avoid deep relative paths when importing modules (e.g. from within a `test/` directory), the `no-extraneous-dependencies` rule is triggered.

I'd like to define a whitelist of allowed "extraneous dependencies" that map to my webpack aliases.

## Suggestion for configuration option

Use a key `"whitelist"` to enumerate dependencies that should be skipped in the check.

```json
{
  "import/no-extraneous-dependencies": [
    "error",
    {
      "devDependencies": false,
      "optionalDependencies": false,
      "peerDependencies": false,
      "whitelist": ["src"]
    }
  ]
```

## Repro

*Webpack config*
```js
...
resolve: {
  extensions: ['', '.js', '.json'],
  alias: {
    src: path.join(process.cwd(), 'src')
  }
},
...
```

*Code*
```js
import MyComponent from 'src/components/MyComponent';
```

**Comment by coopy** [comment]
I figured out that properly configuring [`eslint-import-resolver-webpack`](https://www.npmjs.com/package/eslint-import-resolver-webpack) with the correct path to my webpack configuration resolved the issue! This project has a lot of features, thank you :]
